api/v1alpha4/clusterclass_types.go

```diff
@@ -53,6 +53,20 @@ type ClusterClassSpec struct {
 	Workers WorkersClass `json:"workers,omitempty"`
 }
 
+type ControlPlaneClass struct {
+	Metadata ObjectMeta `json:"metadata,omitempty"`
+
+	// LocalObjectTemplate contains the reference to the control plane provider.
+	LocalObjectTemplate `json:",inline"`
+
+	// MachineTemplate defines the metadata and infrastructure information
+	// for control plane machines.
+	//
+	// This field is supported if and only if the control plane provider template
+	// referenced above is Machine based and supports setting replicas.
+	MachineInfrastructure LocalObjectTemplate `json:"machineInfrastructure"`
```

Comment:
@sbueringer @killianmuldoon we should update the definition of cluster class compatibility considering this field
@ykakarap we should consider this in additional ClusterClass validations